### PR TITLE
Add initial UXML file support

### PR DESCRIPTION
**Pull request by citizenmatt**
This PR adds basic UXML file support:

- [x] Treat UXML files as XML (syntax highlighting, etc.)
- [x] Automatically load and apply XML schema from the `UIElementsSchema` folder in the project root
- [x] Show editor notification panel with link to generate the UIElements schema via the protocol + refresh the XML editor

Known issues:

* The UXML file icon isn't correct - [RIDER-34788](https://youtrack.jetbrains.com/issue/RIDER-34788)
* The `class` and `style` attributes are highlighted as an error. This is a bug in the XML tooling. The schema should accept `anyAttribute` but doesn't - [IDEA-226310](https://youtrack.jetbrains.com/issue/IDEA-226310)

### CODE REVIEW

**Comment by citizenmatt** [comment]
To test:

* Clone/download the [Unity-Technologies/UIElementsExamples](https://github.com/Unity-Technologies/UIElementsExamples) repo
* Open the project in Unity + Rider, edit the `.uxml` files. They should be treated as XML files.
* Open the project in Unity 2018.1 and below
  * UXML is an experimental feature in 2018.2 and above, and officially supported in 2019.1 and above.
  * Open a `.uxml` file in an older Unity project (create a new file if the repo project doesn't open). The editor displays a notification panel saying that "UXML support requires 2019.1 and above". We only recommend the version where it is no longer experimental.
* Open the project in Unity 2018.2 and above
  * Opening a `.uxml` file will prompt to generate the XML schema that will provide validation + code completion.
  * Clicking the "generate schema" link will use the protocol to tell the Unity editor to generate the schema in the `UIElementsSchema` folder in the root of the project.
  * Once the files are generated, they should be automatically loaded and applied to the UXML file. Code completion now works for child elements, e.g. `<ui:Button …>` and attributes
  * The `class` attribute is highlighted as an error due to a bug in the XML tooling. See [IDEA-226310](https://youtrack.jetbrains.com/issue/IDEA-226310)
* Close and reopen the project in Rider
  * The schema files should already be generated, so opening a `.uxml` file should automatically associate the schema and should not display a notification
* Open the project in Rider without starting Unity
  * Make sure the `UIElementsSchema` folder does not exist
  * Opening a `.uxml` file should prompt to generate the schema
  * Clicking the "generate schema" link will start Unity. Once started, the schema should be automatically generated and associated with the files
  * Code completion + validation should work

Note that some files in the repo use the old, unsupported experimental namespaces. These files will not have schemas associated, and will not have code completion, etc.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ideaInterop/fileTypes/uxml/codeInsight/schema/UxmlSchemaProvider.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return schema.value
+    }
+
+    override fun getAvailableNamespaces(file: XmlFile, tagName: String?): MutableSet<String> {
+        // For the given XmlFile, what namespaces do we know about?
+        // The schemas are named after the namespace
+        return getSchemas(file.project).keys
+    }
+
+    override fun getLocations(namespace: String, context: XmlFile): MutableSet<String>? {
```
Comment (kashmervil) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ideaInterop/fileTypes/uxml/codeInsight/schema/UxmlSchemaProvider.kt]: You can restrict the type of the override to `Set<String>?`

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ideaInterop/fileTypes/uxml/codeInsight/schema/UxmlSchemaProvider.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        val project = baseFile.project
+        val schemas = getSchemas(project)
+
+        val cachedValue = schemas[url]
+        if (cachedValue != null) return cachedValue.value
+
+        val schema = CachedValuesManager.getManager(project).createCachedValue(object : CachedValueProvider<XmlFile> {
+            override fun compute(): CachedValueProvider.Result<XmlFile>? {
+                val file = project.projectDir.findFileByRelativePath("/UIElementsSchema/$url.xsd")
```
Comment (kashmervil) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ideaInterop/fileTypes/uxml/codeInsight/schema/UxmlSchemaProvider.kt]: Are you sure it's the best place to put such a file?

Comment (citizenmatt) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ideaInterop/fileTypes/uxml/codeInsight/schema/UxmlSchemaProvider.kt]: That's where Unity puts it. We don't get any choice in location.

File: unity/EditorPlugin/UIElementsSupport.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public static bool GenerateSchema()
+    {
+      // This type was first introduced in 2019.1
+      var generator = Type.GetType("UnityEditor.UIElements.UxmlSchemaGenerator,UnityEditor");
+      if (generator == null)
+      {
+        ourLogger.Verbose("Cannot find type: UnityEditor.UIElements.UxmlSchemaGenerator. Trying obsolete API");
+
+        // The schema generator was first introduced in experimental namespace in 2018.2
+        generator = Type.GetType("UnityEditor.UIElements.UxmlSchemaGenerator,UnityEditor");
```
[the remark this answers is not in the document]
Comment (citizenmatt) [reply unity/EditorPlugin/UIElementsSupport.cs]: Good spot, thanks! Fixed.

**Pull request merged**